Src/Web/Web.Shared.Net/PostSamplingTelemetryProcessor.cs

```diff
@@ -0,0 +1,77 @@
+﻿namespace Microsoft.ApplicationInsights.Web
+{
+    using System;
+
+    using Microsoft.ApplicationInsights.Channel;
+    using Microsoft.ApplicationInsights.Common;
+    using Microsoft.ApplicationInsights.DataContracts;
+    using Microsoft.ApplicationInsights.Extensibility;
+    using Microsoft.ApplicationInsights.Extensibility.Implementation.Tracing;
+    using Microsoft.ApplicationInsights.Web.Implementation;
+
+    /// <summary>
+    /// PostSamplingTelemetry processor which ensures that the RequestTelemetry.Url is initialized
+    /// in the context of HttpContext.Current.Request
+    /// </summary>
+    public class PostSamplingTelemetryProcessor : ITelemetryProcessor
```

Comment:
These are public classes, and will be released as part of regular nugets, so lets ship only those things which we really want. Its not easy to remove once shipped.

Reply:
If it's good enough for Attach customers, it should be good enough for all SDK customers  :)

What would be required for this to be a part of the full SDK?

Reply:
URL is already populated by RequestCollectionModules which is shipped officially.
Modifying existing RequestCollectionModules to NOT populate URL, and instead rely on this new TP is a behavior change- Tel.Initializers will not see the URL. And also any TP added before this TP wont see URL. And there is no guarantee that TP is run in same context as incoming Request, so HttpContext.Current could be null.

We can discuss such change, but i'd do it only when we are changing major version.

Reply:
That's a good point. I'm going to close this PR and open an issue to discuss further.

Reply:
https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/1173